pgmpy/continuous/discretize/base.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import division
+
+from six import with_metaclass
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+from pgmpy.factors import ContinuousFactor
+
+
+class BaseDiscretizer(with_metaclass(ABCMeta)):
```

Comment:
@ankurankan I have used the abc module to implement the abstract method ```get_discrete_values```. It requires using a metaclass ```abc.ABCMeta```. Since metaclasses are used differently in python 2 and 3, I used ```six.with_metaclass```. An another way to do this would be using a decorator ```@add_metaclass(ABCMeta)``` on the ```BaseDiscretizer``` class.
If there is a better way to do this, please let me know. I will change it.